### DISCUSSION TITLE
Demo Module Razor Pages path documentation

### DISCUSSION THREAD

**Discussion by JoshTango** [OrchardCMS/OrchardCore#12755] [General]
In the demo module there are some razor pages and in the startup some mappings.  Can you please explain the url for these Razor pages?

**Comment by Skrypt** [comment]
Each OC module has an area name equal to that module name. So, Pages are following that same logic per module.

**Comment by JoshTango** [comment]
I did get it to work eventually through trial and error.

**Comment by ns8482e** [comment]
Mapping razor pages are easy ones. You can  define them using page route e.g. `@page "~/route" `  
You can do more advance routing using `Startup` as shown in Demo module

**Comment by JoshTango** [comment]
This "Each OC module has an area name equal to that module name." should probably be put into the documentation somewhere.

**Comment by ns8482e** [comment]
@JoshTango  it's already documented here 

> By default, all routes in modules follow the pattern {area}/{controller}/{action}, where {area} is the name of the module.

https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/#registering-a-custom-route

**Comment by JoshTango** [comment]
Ok thanks.  It would be nice if there was a Razor Pages section with that exact documentation.

**Comment by JoshTango** [comment]
I went searching for Razor pages and didnt find a section.

**Comment by JoshTango** [comment]
When I typed in the search box all I saw was Razor Helpers.  So maybe have that document that @ns8482e  linked to show up.
